torch/_functorch/eager_transforms.py

```diff
@@ -60,6 +60,17 @@ def enable_inplace_requires_grad(enabled):
         set_inplace_requires_grad_allowed(prev_state)
 
 
+def _vjp_treespec_compare(primals_out, cotangents):
+    _, primals_out_spec = tree_flatten(primals_out)
+    _, cotangents_spec = tree_flatten(cotangents)
+    if primals_out_spec != cotangents_spec:
+        raise RuntimeError(
+            f'Expected pytree structure of cotangents to be the same '
+            f'as pytree structure of outputs to the function. '
+            f'cotangents: {treespec_pprint(cotangents_spec)}, '
+            f'primal output: {treespec_pprint(primals_out_spec)}')
```

Comment:
Why do we need to move this to a separate function and flatten `primals_out` and `cotangents` again?